### PR TITLE
Make output directories of RobotImporter unique

### PR DESCRIPTION
**Pull request by lewandowskikacper**
Added Uuid of parameters passed to XACRO to the name name of RobotImporter's output directory.
This makes the directory unique for each robot imported from the same xacro file. 
Addresses  https://github.com/o3de/o3de-extras/issues/392.

### CODE REVIEW

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h [hunk 1]
```diff
@@ -119,6 +119,7 @@ namespace ROS2::Utils
     UrdfAssetMap CopyAssetForURDFAndCreateAssetMap(
         const AZStd::unordered_set<AZStd::string>& meshesFilenames,
         const AZStd::string& urdfFilename,
+        AZ::Crc32 params_crc,
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h:122]: These parameters need to be documented. I do not like it CRC32. Please consider adding an optional parameter that extends/overrides the target directory name.

File: Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp [hunk 2]
```diff
@@ -181,10 +182,16 @@ namespace ROS2
             auto collidersNames = Utils::GetMeshesFilenames(m_parsedUrdf->getRoot(), false, true);
             auto visualNames = Utils::GetMeshesFilenames(m_parsedUrdf->getRoot(), true, false);
 
+            AZ::Crc32 params_crc;
+            for (auto& [key, value] : m_params)
+            {
+                params_crc.Add(key + "_" + value);
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp:188]:
```suggestion
                params_crc.Add(key)
                params_crc.Add(value);
```

**Review by michalpelka — COMMENTED** [review]
Please adjust changes to CopyAssetForURDFAndCreateAssetMap

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h (hunk 1 above)
Comment (lewandowskikacper) [reply Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.h]: changed to optional parameter OutputDirSuffx of type AZStd::string

File: Gems/ROS2/Code/Source/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp [hunk 3]
```diff
@@ -7,6 +7,7 @@
  */
 
 #include "ROS2RobotImporterEditorSystemComponent.h"
+#include "AzCore/Math/Crc.h"
```
Comment (lumberyard-employee-dm) [inline Gems/ROS2/Code/Source/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp:10]: This include is unused and should be removed.

File: Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp [hunk 4]
```diff
@@ -181,10 +182,19 @@ namespace ROS2
...
+                params_crc.Add(value);
+            }
+
+            auto dir_suffix = AZStd::string::format("%d", AZ::u32(params_crc));
+
             if (m_importAssetWithUrdf)
             {
                 m_urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>(
-                    Utils::CopyAssetForURDFAndCreateAssetMap(m_meshNames, m_urdfPath.String(), collidersNames, visualNames));
+                    Utils::CopyAssetForURDFAndCreateAssetMap(m_meshNames, m_urdfPath.String(), collidersNames, visualNames, dir_suffix));
```
Comment (lumberyard-employee-dm) [inline Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp:197]: The naming scheme of the `dir_suffix` parameter is inconsistent with the other parameters and should be camel case per the [coding standard](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#naming).

i.e the name could be `dirSuffix`

File: Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp [hunk 5]
```diff
@@ -181,10 +182,19 @@ namespace ROS2
             auto collidersNames = Utils::GetMeshesFilenames(m_parsedUrdf->getRoot(), false, true);
             auto visualNames = Utils::GetMeshesFilenames(m_parsedUrdf->getRoot(), true, false);
 
+            AZ::Crc32 params_crc;
+            for (auto& [key, value] : m_params)
+            {
+                params_crc.Add(key);
+                params_crc.Add(value);
+            }
```
Comment (lumberyard-employee-dm) [inline Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp:190]: The [AZ::Uuid::CreateName](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/Uuid.h#L77-L78) function to create a Uuid from a string.
The [Uuid::operator+](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Math/Uuid.h#L143-L144) can then be used to combine the Uuid of two strings

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp [hunk 6]
```diff
@@ -255,6 +256,7 @@ namespace ROS2::Utils
         const AZStd::string& urdfFilename,
         const AZStd::unordered_set<AZStd::string>& colliders,
         const AZStd::unordered_set<AZStd::string>& visuals,
+        const AZStd::string& outputDirSuffix,
```
Comment (lumberyard-employee-dm) [inline Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp:259]: This parameter can be an `AZStd::string_view` which is more flexible than an `const AZStd::string&` parameter and avoids temporary heap allocations when passing in a string literal.

File: Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp [hunk 7]
```diff
@@ -274,7 +276,8 @@ namespace ROS2::Utils
         urdfFileCrc.Add(urdfFilename);
         const AZ::IO::Path urdfPath(urdfFilename);
         const AZStd::string directoryNameTmp = AZStd::string::format("$tmp_%u.tmp", AZ::u32(urdfFileCrc));
-        const AZStd::string directoryNameDst = AZStd::string::format("%u_%s", AZ::u32(urdfFileCrc), urdfPath.Stem().String().c_str());
+        const AZStd::string directoryNameDst =
+            AZStd::string::format("%u_%s%s", AZ::u32(urdfFileCrc), urdfPath.Stem().String().c_str(), outputDirSuffix.c_str());
```
Comment (lumberyard-employee-dm) [inline Gems/ROS2/Code/Source/RobotImporter/Utils/SourceAssetsStorage.cpp:280]: This format can be updated to format the suffix from the string view.
Also using the string type of "AZ::IO::FixedMaxPathString" which is a type alias of  `AZStd::fixed_string<1024>` avoids a heap allocation here.
Information about fixed_string is documented in its [header file](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/string/fixed_string.h#L20-L25)
```suggestion
        const auto directoryNameDst =
            AZ::IO::FixedMaxPathString::format("%u_%.*s%.*s", AZ::u32(urdfFileCrc), AZ_PATH_ARG(urdfPath.Stem()), AZ_STRING_ARG(outputDirSuffix));
```

**Review by adamdbrw — APPROVED** [review]

**Review by arturkamieniecki — APPROVED** [review]

**Review by michalpelka — APPROVED** [review]

**Pull request merged**